doc/man3/flux_msg_handler_addvec.adoc

```diff
@@ -49,6 +49,17 @@ handler, all previously created message handlers in the array are destroyed
 before an error is returned.
 
 
+SECURITY
+--------
+
+The _rolemask_ member of _struct flux_msg_handler_spec_ is passed
+to `flux_msg_handler_allow_rolemask()`.  If only FLUX_ROLE_OWNER should
+be allowed to send messages to the handler, set _rolemask_ to zero.
```

Comment:
I find this very confusing.  FLUX_ROLE_NONE  is 0, and FLUX_ROLE_OWNER is 1.  So essentially you are saying that if we want to allow FLUX_ROLE_OWNER, we should set FLUX_ROLE_NONE.

I'm not a fan of using zero in the code when we have explicitly set a name for that, FLUX_ROLE_NONE.  So at the least, I would prefer to see "zero" changed to "FLUX_ROLE_NONE".

I read the flux_msg_handler_create(3) man page as instructed below, and it says:

"A message handler only receives a message if a bit-wise AND of the message
rolemask and the message handler rolemask evaluates to a non-zero value.
By default, the message handler rolemask contains only FLUX_ROLE_OWNER
(the instance owner) but may be changed with
`flux_msg_handler_allow_rolemask()` and `flux_msg_handler_deny_rolemask()`,
which add or remove roles from the message handler rolemask.  FLUX_ROLE_OWNER
cannot be dropped from the message handler rolemask."

It is pretty unclear about what happens when we try to set the role to FLUX_ROLE_NONE, despite that being exactly what is instructed in this man page.  I'm assuming that it means that FLUX_ROLE_NONE is ignored and the default value of FLUX_ROLE_OWNER remains.

As I can tell from the documentation, at current time there is no practical difference between FLUX_ROLE_NONE and FLUX_ROLE_OWNER.  But perhaps in the future there will be? (Otherwise, why have them both?)  So perhaps better than setting it to 0/FLUX_ROLE_NONE, it would be better to instruct people to set FLUX_ROLE_OWNER if they want FLUX_ROLE_OWNER.

Reply:
I'm pondering this one.

In general, I don't see a problem with 0 as a value for a bitmask as opposed to a symbolic value.  FLUX_ROLE_NONE was not really intended to be used here; it was defined to be used as the initial value for the rolemask field in a message, e.g. from [RFC 11](https://github.com/flux-framework/rfc/blob/master/spec_12.adoc)

> FLUX_ROLE_NONE (0) SHALL indicate "invalid rolemask".

Possibly that was a poor naming choice.

I think you are right that the way FLUX_ROLE_OWNER as described is confusing.  It cannot be denied and is meaningless to allow, so perhaps instead of being described as implicitly set in the "allow mask" of a handler, it should be an error to include it in the mask or to attempt to allow/deny it.  It would be unfortunate to give the impression that the owner can be denied, since the fact that the owner is all powerful is a fundamental part of the security design.

In any case this pr is about something else so maybe the best thing is to drop the man page update from this PR and do another.